File: 200-299/209. Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int sum=0;
        int res = INT_MAX;
        int left = 0;
        for(int i=0; i<nums.size(); i++){
            sum+=nums[i];
            while(sum>=s){
                res = min(res, i+1-left);
                sum-=nums[left];
                left++;
            }
        }
        return res==INT_MAX ? 0 : res;
    }
};

2019.09.02

File: 2025/209. Minimum Size Subarray Sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        right = 0
        total_sum = 0
        min_length = float("inf")
        
        while left <= right and right < len(nums):
            total_sum += nums[right]
            while total_sum >= target and left <= right:
                min_length = min(min_length, right - left + 1)
                total_sum -= nums[left]
                left += 1
            
            right += 1
        
        if min_length < float("inf"):
            return min_length
        return 0
